peps/pep-0810.rst

```diff
@@ -0,0 +1,1345 @@
+  ``importlib.import_module()``.
+* Import hooks and loaders continue to run under the standard import protocol
+  when a lazy object is reified.
+
+Observable behavioral shifts (opt-in only)
+------------------------------------------
+
+These changes are limited to bindings explicitly made lazy:
+
+* **Error timing.** Exceptions that would have occurred during an eager import
+  (for example ``ImportError`` or ``AttributeError`` for a missing member) now
+  occur at the first *use* of the lazy name.
+
+  .. code-block:: python
+
+    # With eager import - error at import statement
+    import broken_module  # ImportError raised here
+
+    # With lazy import - error deferred
+    lazy import broken_module
+    print("Import succeeded")
+    broken_module.foo()  # ImportError raised here on first use
+
+* **Side-effect timing.** Import-time side effects in lazily imported modules
+  occur at first use of the binding, not at module import time.
+* **Import order.** Because modules are imported on first use, the order in
+  which modules are imported may differ from how they appear in code.
+* **Presence in ``sys.modules``.** A lazily imported module does not appear in
+  ``sys.modules`` until first use. After reification, it must appear in
+  ``sys.modules``. If some other code eagerly imports the same module before
+  first use, the lazy binding resolves to that existing (lazy) module object when
+  it is first used.
+* **Proxy visibility.** Before first use, the bound name refers to a lazy proxy.
+  Indirect introspection that touches the value may observe a proxy lazy object
+  representation. After first use, the name is rebound to the real object and
+  becomes indistinguishable from an eager import.
+
+Thread-safety and reification
+-----------------------------
+
+First use of a lazy binding follows the existing import-lock discipline. Exactly
+one thread performs the import and **atomically rebinds** the importing module's
+global to the resolved object. Concurrent readers thereafter observe the real
+object.
+
+Lazy imports are thread-safe and have no special considerations for free-threading.
+A module that would normally be imported in the main thread may be imported in a
+different thread if that thread triggers the first access to the lazy import. This
+is not a problem: the import lock ensures thread safety regardless of which thread
+performs the import.
+
+Subinterpreters are supported. Each subinterpreter maintains its own
+``sys.lazy_modules`` and import state, so lazy imports in one subinterpreter do
+not affect others.
+
+Typing and tools
+----------------
+
+Type checkers and static analyzers may treat ``lazy`` imports as ordinary
+imports for name resolution. At runtime, annotation-only imports can be marked
+``lazy`` to avoid startup overhead. IDEs and debuggers should be prepared to
+display lazy proxies before first use and the real objects thereafter.
+
+Lazy imports filter
```

Comment:
This is in the "Backwards Compatibility" section but shouldn't be, since it proposes a new feature. Move it to Specification.